stonesoup/sensor/base.py

```diff
@@ -1,101 +1,103 @@
-        return self._platform_system
-
-    @platform_system.setter
-    def platform_system(self, value: weakref.ref):
-        # this slightly odd construction is to allow overriding by the Sensor subclass
-        self._set_platform_system(value)
-
-    def _set_platform_system(self, value: weakref.ref):
-        if self._platform_system is not None:
-            warn('Sensor has been moved from one platform to another. This is unexpected '
-                 'behaviour')
-        self._platform_system = value
-
-    @abstractmethod
-    def measure(self, ground_truths: Set[GroundTruthState], noise: Union[np.ndarray, bool] = True,
-                **kwargs) -> Set[TrueDetection]:
-        """Generate a measurement for a given state
-
-        Parameters
-        ----------
-        ground_truths : Set[:class:`~.GroundTruthState`]
-            A set of :class:`~.GroundTruthState`
-        noise: :class:`numpy.ndarray` or bool
-            An externally generated random process noise sample (the default is `True`, in which
-            case :meth:`~.Model.rvs` is used; if `False`, no noise will be added)
-
-        Returns
-        -------
-        Set[:class:`~.TrueDetection`]
-            A set of measurements generated from the given states. The timestamps of the
-            measurements are set equal to that of the corresponding states that they were
-            calculated from. Each measurement stores the ground truth path that it was produced
-            from.
-        """
-        raise NotImplementedError
+        if position is not None or orientation is not None:
+            if position is None:
+                # assuming 3d for a default platform
+                position = StateVector([0, 0, 0])
+            if orientation is None:
+                orientation = StateVector([0, 0, 0])
+            self._internal_movement_controller = FixedMovable(
+                states=State(state_vector=position),
+                position_mapping=list(range(len(position))),
+                orientation=orientation)
+            self._property_movement_controller = self._internal_movement_controller
+            self._set_mounting_rotation_defaults()
+
+    def _set_mounting_rotation_defaults(self):
+        if self.movement_controller is None:
+            return
+        if self.mounting_offset is None:
+            self.mounting_offset = (StateVector([0]
+                                    * len(self.movement_controller.position_mapping)))
+
+        if self.rotation_offset is None:
+            self.rotation_offset = StateVector([0] * 3)
+
+    @movement_controller.setter
+    def movement_controller(self, value):
+        if self._has_internal_controller:
+            raise AttributeError('The movement controller cannot be set on a Sensor with a private'
+                                 'internal controller')
+        self._property_movement_controller = value
```

Comment:
Should this not be the attribute that is checked against later in the code, rather than `_internal_movement_controller`?
I'm confused over the use of having both.

Reply:
You did notice an oddity here. I typed up the explanation below, before looking at the code and realizing it didn't actually do that! I have reworked the code and I'll try and explain the new logic here:

If no moveable is supplied to a sensor at construction, but it is given a position and/or orientation, it constructs an _internal_ `FixedMovable` to handle it's position. This is stored in the `movement_controller` Property and cannot be changed during the life of the object. It allows the sensor position and orientation to be set.

The `_has_internal_controller` is a (readonly) internal boolean flag that is set to indicate this state. It cannot be stored in the `movement_controller` Property as this used for either a "normal" controller or an internal one.